index.js

```diff
@@ -138,3 +138,41 @@ newSightingForm.addEventListener('submit', function (event) {
 /************************** EVENTS PART 2 JS MINI CHALLENGE ******************************/
 
 
+//Deliv 1 & 2
+
+const dataIdLi = document.querySelector('ul#animals')
+
+dataIdLi.addEventListener('click', function(event){
+    let sighting = event.target.closest('li')
+    if (event.target.matches('button.like-button')){
+        const likesPtag = event.target.previousElementSibling
+        const likes = parseInt(likesPtag.textContent) + 1
+        likesPtag.textContent = `${likes} Likes`
+    } else if (event.target.matches('button.delete-button')){
+        sighting.remove()
+    } else if (event.target.matches('button.toggle-update-form-button')){
+       let updateForm = event.target.nextElementSibling
+        if (updateForm.style.display === 'none'){
+            updateForm.style.display = 'block'
+        } else {
+            updateForm.style.display = 'none'
+        }
+    }
+})
```

Comment:
Nice! You have an event listener on the entire animals UL element, and based on the class names you are delegating out different logic. Great. First step is getting everything up and running. Awesome. Next, we can refactor for readability and to make our code more reusable.

```
/***** DELIVERABLE 1, 2, & 3 part 1 *****/

const animalsUL = document.querySelector('ul#animals')
animalsUL.addEventListener("click", function(event){
    if (event.target.matches('button.like-button')){
        handleLike(event)
    } else if (event.target.matches('button.delete-button')){
        handleDelete(event)
    } else if (event.target.matches('button.toggle-update-form-button')){
        handleToggleUpdate(event)
    }
})

/***** DELIVERABLE 1 part 2 *****/
function handleLike(e){
    const likesDisplayPTag = e.target.previousElementSibling
    const currentTextContent = likesDisplayPTag.innerText
    let numberOfLikes = parseInt(currentTextContent.split(" ")[0])
    numberOfLikes += 1
    const newTextContent = `${numberOfLikes} Likes`
    likesDisplayPTag.innerText = newTextContent
}

/***** DELIVERABLE 2 part 2 *****/
function handleDelete(e){
    const parentLI = e.target.parentElement
    parentLI.remove()
}

/***** DELIVERABLE 3 part 2 *****/
function handleToggleUpdate(e){
    const updateForm = e.target.nextElementSibling
    toggleHidden(updateForm)
}

/***** DELIVERABLE 3 part 3 *****/
function toggleHidden(element){
    element.style.display === "none" ? element.style.display = "block" : element.style.display = "none"
}
```

Reply:
As you can see I am adding a few extra steps in the `handleLike` function. Some of the steps are unnecessary but for me I want to make sure it's easy for anyone to read this code and understand what each variable is holding onto and its purpose.